libraries/eosiolib/contracts/eosio/key_value.hpp

```diff
@@ -0,0 +1,1051 @@
+#pragma once
+#include "../../core/eosio/datastream.hpp"
+#include "../../core/eosio/name.hpp"
+#include "../../core/eosio/print.hpp"
+#include "../../core/eosio/utility.hpp"
+
+#include <algorithm>
+#include <cctype>
+#include <functional>
+
+#include <boost/preprocessor/variadic/to_seq.hpp>
+#include <boost/preprocessor/seq/for_each_i.hpp>
+#include <boost/pfr.hpp>
+
+#define EOSIO_CDT_KV_INDEX_nullptr
```

Comment:
TEST_1 causes a compilation error, though I'm not sure how to resolve that. TEST works fine. I'm not sure exactly which `_` you're referring to, nothing is being prefixed with a `_` so we shouldn't run into an `__` names.

Reply:
Below you have `EOSIO_CDT_KV_INDEX_ ## index_name`  This must not expand to either:
a) a reserved identifier  (any name containing a double underscore is reserved to the implementation for any use)
b) a macro that you defined that it isn't intended to match.
Since `index_name` is user-defined, the only constraint on it is that it must not be a reserved name.  The fix for both is to change the prefix `EOSIO_CDT_KV_INDEX_`